src/types/job.rs

```diff
@@ -33,7 +33,10 @@ pub struct JobDescriptor {
     pub pass: bool,
     pub msg: String,
     pub date: String,
+    // don't deprecate until `ecosystems` is live.
+    //#[deprecated = "Use `ecosystems` to support multiple ecosystems."]
     pub ecosystem: String,
```

Comment:
We should make this value optional for deserialization so that CLI versions released during this transition will continue to work after the transition (i.e., when this value is completely removed). The easiest way to do that is to just use `#[serde(default)]`.
```suggestion
    // don't deprecate until `ecosystems` is live.
    //#[deprecated = "Use `ecosystems` to support multiple ecosystems."]
    #[serde(default)]
    pub ecosystem: String,
```

Reply:
This value is not going away so it doesn't need a default.

Reply:
It is going away eventually, right? That's what `#[deprecated]` implies...

I guess we can wait to mark it as default and deprecated at the same time.

Reply:
Maybe. For versioned APIs we don't want to remove things once they've been made available.